packages/analytics/src/Analytics.ts

```diff
@@ -113,17 +114,22 @@ class Analytics {
    *
    * @param key - Key to retrieve from the context. If not specified, will return the whole data stored
    *              in the context.
+   * @param eventType - The type of event being tracked. Ex: page, track, onSetUser.
    *
    * @returns Value for the key in context or the whole context data if key is not specified.
    */
   context(): Promise<ContextData>;
-  context(key: string): Promise<unknown>;
-  async context(key?: string) {
+  context(key: string, eventType: ExtendedTrackTypes): Promise<unknown>;
+  context(
+    key: string | undefined,
+    eventType?: ExtendedTrackTypes,
+  ): Promise<unknown>;
+  async context(key?: string, eventType?: ExtendedTrackTypes) {
     const externalContextData = {};
 
     for (const contextFn of this.contextFns) {
       try {
-        const contextFnResult = await contextFn();
+        const contextFnResult = await contextFn(eventType);
```

Comment:
Question: Why is it necessary to support different contexts by event type? Is there a specific case where you need to provide different contexts for different event types?